src/services/AuthService.js

```diff
@@ -0,0 +1,105 @@
+import config from "@/config"
+import OauthAPI from "./api/OauthAPI"
+
+const api = new OauthAPI()
+
+export default class AuthService {
+  constructor () {
+    this.clientSecret = config.datagouvfr_oauth_client_secret
+    this.clientId = config.datagouvfr_oauth_client_id
+    this.baseURL = config.datagouvfr_base_url
+    this.redirectURI = `${window.location.origin}/login/callback`
+  }
+
+  /**
+   * Compute data.gouv.fr URL to be redirected to for oauth authorization
+   */
+  async getRedirectURL () {
+    const codeVerifier = this.generateRandomString()
+    const codeChallenge = await this.pkceChallengeFromVerifier(codeVerifier)
+    const state = this.generateRandomString()
+    const redirectURI = encodeURIComponent(this.redirectURI)
+    const encodedState = encodeURIComponent(state)
+    const encodedCC = encodeURIComponent(codeChallenge)
+
+    localStorage.setItem("pkceCodeVerifier", codeVerifier)
+    localStorage.setItem("pkceState", state)
+
+    return `${this.baseURL}/oauth/authorize?redirect_uri=${redirectURI}&response_type=code&state=${encodedState}&client_id=${this.clientId}&scope=default&code_challenge=${encodedCC}&code_challenge_method=S256`
```

Comment:
The data.gouv auth page displays:
> client-01 voudrait accéder à votre compte demo.data.gouv.fr.

Is it possible to make it display the ecosphères URL (or at least "ecosphères") instead of "client-01"?

Reply:
Yes, it's a a value in DB on data.gouv.fr's side, we can ask them.